source/lib/tabs-service.js

```diff
@@ -36,30 +36,30 @@ export async function queryTabs(urlList) {
 export async function openTab(url) {
 	const {newTabAlways} = await syncStore.getAll();
 
-	if (!newTabAlways) {
-		const alreadyGranted = await queryPermission('tabs');
-		if (!alreadyGranted) {
-			const granted = await requestPermission('tabs');
-			if (!granted) {
-				return;
-			}
-		}
+	if (newTabAlways) {
+		return createTab(url);
+	}
 
-		const matchingUrls = [url];
-		if (url.endsWith('/notifications')) {
-			matchingUrls.push(url + '?all=1');
+	const alreadyGranted = await queryPermission('tabs');
```

Comment:
The mistake here is asking for permission “at runtime” instead of when the user checks the box. This check/request shouldn’t be here. 

The option itself seems backwards, it should be “Reuse notifications tabs when possible: default to false”

When the user checks this, the permission is requested. If the permission is denied we need to uncheck the box

Reply:
> The mistake here is asking for permission “at runtime” instead of when the user checks the box. This check/request shouldn’t be here.

`openTab` almost always happens as part of a user input. According to the second example mentioned [here](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/User_actions), there cannot be any promises waited up before calling `browser.permissions.request()` (in Firefox only, Chrome is permissive).

I'm sure people at Firefox had a reason to do this, but I don't see how to solve this other than to revert #162, so we can drop `await` from [here](https://github.com/sindresorhus/notifier-for-github/blob/00d53edecf7587bd462fb8fc6b1716fc05b5eaeb/source/lib/tabs-service.js#L25).

------------------------

> The option itself seems backwards, it should be “Reuse notifications tabs when possible: default to false”

As mentioned, #162 can be reverted, @sindresorhus WDYT!?

------------------------

> When the user checks this, the permission is requested. If the permission is denied we need to uncheck the box

That already worked in Chrome, with https://github.com/sindresorhus/notifier-for-github/pull/180/commits/00d53edecf7587bd462fb8fc6b1716fc05b5eaeb, it now works in Firefox too.

Reply:
The solution is not dropping it but following what I just said: don’t ask permissions when opening the tab

Reply:
The only place we request `tabs` permission is here 😕. Not asking permission here _or_ dropping #162, either of them require us to move `tabs` to required permissions (which is what I'm inclined towards, and need other people' opinion before making this change).

Reply:
`tabs` should stay optional. There are lots of people that would never want to allow it as it gives full access to all browsing history.

Reply:
Let's go with @bfred-it's suggestion then, to modify the existing option.

Reply:
If you want to edits at a minimum, you can leave the current option as is and just move the `permissions.request` to the options page.